### ISSUE TITLE
Multi word match query minShould

### ISSUE THREAD

**Issue by jgschis** [blevesearch/bleve#1747]
Hi,

let's say a user enters 5 words into the search bar. I want to return all documents that contain at least 4 of the words. How do I do that in Bleve? It seems that Bleve only has 2 options:

1. any term (or) 
2. all terms (and) 

but nothing in between. 

Btw, Elasticearch can do this:
https://www.elastic.co/guide/en/elasticsearch/guide/current/match-multi-word.html#match-improving-precision

**Comment by abhinavdangeti** [comment]
You'll want to use a disjunction query with a `min` for this. Here's a JSON example -

```
{
  "query": {
    "disjuncts": [
      {"match": "one"},
      {"match": "two"},
      {"match": "three"},
      {"match": "four"},
      {"match": "five"}
    ],
    "min": 4
  }
}
```

If you were using a query string, doing this would look for any of the 5 words ..
```
"one two three four five"
```
.. **but** will now allow you to set a value for "min".

**Comment by abhinavdangeti** [comment]
One more thing, the elasticsearch section you highlight is supported within bleve as well.

This query ..
```
{
  "query": {
    "match": "brown dog",
    "field": "title"
  }
}
```
.. will look for the occurrence of either `brown` or `dog` in the field **title** (assuming you're using the standard analyzer).
And the following query will look for both `brown` and `dog` in the field **title**:
```
{
  "query": {
    "match": "brown dog",
    "field": "title",
    "operator": "and"
  }
}
```

**Comment by jgschis** [comment]
The problem with what you are saying is that I have to split the user's search query into sub queries myself. But the search-time anayser should be doing it,  as only it knows how to correctly do the splitting.  I can assume that the query should be slit on whitespace, but this will not always be true -- it depends on the analyser.

**Comment by abhinavdangeti** [comment]
Yes, an analyzer is responsible for tokenizing (on unicode or whitespace or something else), stemming, replacing and/or applying other filtering to text. This is applicable at index time and also at query time but only for analytic queries - `match`, `match_phrase`.

However, an analyzer will not handle an at-least-these-many-matches situation. 

The matching precision offered by elastic is not directly supported by bleve at the moment (we'll consider supporting it for a future release). The current workaround is to use a disjunction query with the `min` setting.

**Comment by jgschis** [comment]
I could look into doing this...I will.
